HippUnfoldHCP/HippUnfoldHCP.sh

```diff
@@ -0,0 +1,91 @@
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+
+
+opts_SetScriptDescription "Make some BIDS structures and run HippUnfold"
+
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder containing all subjects"
+opts_AddMandatory '--subject' 'Subject' 'subject ID' ""
+opts_AddOptional '--hippunfold-dir' 'HippUnfoldDIR' 'path' "location of HippUnfold outputs"
+
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+opts_ShowValues
+
+T1wFolder="$StudyFolder/$Subject/T1w"       # input data
+
+if [ -z ${HippUnfoldDIR} ] ; then
+  HippUnfoldDIR="${T1wFolder}/HippUnfold"
+fi
+
+HippUnfoldT1wFolder="$HippUnfoldDIR/T1w/hippunfold"
+HippUnfoldT2wFolder="$HippUnfoldDIR/T2w/hippunfold"
+HippUnfoldT1wT2wFolder="$HippUnfoldDIR/T1wT2w/hippunfold"
+
+T1wImage="$T1wFolder/T1w_acpc_dc_restore.nii.gz"
+T2wImage="$T1wFolder/T2w_acpc_dc_restore.nii.gz"
+
+
+if [ ! -f "$T1wImage" ]; then
+    echo "Error: T1w image not found at $T1wImage" >&2
+    exit 1
+fi
+
+if [ ! -f "$T2wImage" ]; then
+    echo "Error: T2w image not found at $T2wImage" >&2
+    exit 1
+fi
+
+
+mkdir -p "$HippUnfoldT1wFolder" "$HippUnfoldT2wFolder" "$HippUnfoldT1wT2wFolder"
+
+ln -sf "$T1wImage" "$HippUnfoldT1wFolder/s_${Subject}_T1w_acpc_dc_restore.nii.gz"
+ln -sf "$T2wImage" "$HippUnfoldT1wFolder/s_${Subject}_T2w_acpc_dc_restore.nii.gz"
+ln -sf "$T1wImage" "$HippUnfoldT2wFolder/s_${Subject}_T1w_acpc_dc_restore.nii.gz"
+ln -sf "$T2wImage" "$HippUnfoldT2wFolder/s_${Subject}_T2w_acpc_dc_restore.nii.gz"
+ln -sf "$T1wImage" "$HippUnfoldT1wT2wFolder/s_${Subject}_T1w_acpc_dc_restore.nii.gz"
+ln -sf "$T2wImage" "$HippUnfoldT1wT2wFolder/s_${Subject}_T2w_acpc_dc_restore.nii.gz"
+
+log_Msg "Created folder structure under $HippUnfoldDIR and copied T1w and T2w images"
+log_Msg "Starting HippUnfold pipeline for subject: $Subject"
+
+log_Msg "Running T1w HippUnfold for subject: $Subject"                                                                                                      #Seriously: don't put a $ here and don't capitalize S in subject...    #Seriously: don't put a $ here and don't capitalize S in subject...
+if [ -z ${HIPPUNFOLDPATH} ] ; then
+  hippunfold $HippUnfoldT1wFolder $HippUnfoldT1wFolder participant --modality T1w --path-T1w $HippUnfoldT1wFolder/s_{subject}_T1w_acpc_dc_restore.nii.gz --path-T2w $HippUnfoldT1wFolder/s_{subject}_T2w_acpc_dc_restore.nii.gz --cores all --force-output --generate_myelin_map --output-density native 512 2k 8k 18k --use-conda
+else
+  apptainer run --bind $StudyFolder -e $HIPPUNFOLDPATH $HippUnfoldT1wFolder $HippUnfoldT1wFolder participant --modality T1w --path-T1w $HippUnfoldT1wFolder/s_{subject}_T1w_acpc_dc_restore.nii.gz --path-T2w $HippUnfoldT1wFolder/s_{subject}_T2w_acpc_dc_restore.nii.gz --cores all --force-output --generate_myelin_map --output-density native 512 2k 8k 18k
+fi
```

Comment:
If `--use-conda` doesn't need to be the last argument, then we should probably this kind of construct instead, so that we don't accidentally desync the arguments between container vs conda:

```
if [[ "${HIPPUNFOLDPATH:-}" == "" ]]
then
    hippcmd=(hippunfold --use-conda)
else
    hippcmd=(apptainer run --bind "$StudyFolder" -e "$HIPPUNFOLDPATH")
fi
"${hippcmd[@]}" "$HippUnfoldT1wFolder" "$HippUnfoldT1wFolder" participant --modality T1w --path-T1w "$HippUnfoldT1wFolder"/s_{subject}_T1w_acpc_dc_restore.nii.gz --path-T2w "$HippUnfoldT1wFolder"/s_{subject}_T2w_acpc_dc_restore.nii.gz --cores all --force-output --generate_myelin_map --output-density native 512 2k 8k 18k
"${hippcmd[@]}" <next command's arguments>
...
```

Reply:
I agree, feel free to apply this edit over my latest changes. I think the script is not yet in a final state as there are also still some "cleanup" comments in there (`#Seriously: don't put a $ here and don't capitalize S in subject...`).

Reply:
I have made this edit.  I also noticed that the conditional for specifically the `T1T2w` nnunet command looked backwards (ran `apptainer` when the container path variable was empty, and vice versa).  Why did that pass your testing?

Reply:
Tests were ran two times, once by using apptainer and once with hippunfold by commenting out the code. If statements were added post-festum. I plan to retest everything once we have official hippunfold 2.0 and a new wb_command and after that we can merge into master, both here and my interface on the QuNex side. So these were not final tests yet.